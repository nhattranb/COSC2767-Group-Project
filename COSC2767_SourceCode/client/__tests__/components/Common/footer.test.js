/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 17 Dec 2024
 *   Last modified: 17 Dec 2024
 */

import React from 'react';
import {render} from '@testing-library/react';
import Footer from "../../../app/components/Common/Footer";
import {beforeEach, describe, expect, test, jest} from "@jest/globals";
import {Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import {Provider} from "react-redux";
import {combineReducers, createStore} from "redux";


export function createTestStore() {

    // mock reducers
    const userReducer = (state = {}, action) => {
        switch (action.type) {
            default:
                return state;
        }
    }

    const configReducer = (state = {}, action) => {
        switch (action.type) {
            default:
                return state;
        }
    }

    const newsletterReducer = (state = { email: 'test@example.com', formErrors: { email: 'Invalid email address' } }, action) => {
        switch (action.type) {
            default:
                return state;
        }
    }
    return createStore(
        combineReducers({
            user: userReducer,
            config: configReducer,
            newsletter: newsletterReducer
        })
    );
}
describe('Footer component', () => {

    let mockStore;
    beforeEach(() => {
        mockStore = createTestStore();
    });
    test('Footer component matches snapshot', () => {
        const history = createMemoryHistory();

        const { asFragment } = render(
            <Provider store={mockStore}>
                <Router history={history}>
                    <Footer />
                </Router>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    })
});
