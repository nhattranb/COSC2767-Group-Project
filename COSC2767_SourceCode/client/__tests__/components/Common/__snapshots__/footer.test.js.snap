// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="footer-content"
      >
        <div
          class="footer-block"
        >
          <div
            class="block-title"
          >
            <h3
              class="text-uppercase"
            >
              Support
            </h3>
          </div>
          <div
            class="block-content"
          >
            <ul>
              <li
                class="footer-link"
              >
                <a
                  href="/contact"
                >
                  Get in Touch
                </a>
              </li>
              <li
                class="footer-link"
              >
                <a
                  href="/sell"
                >
                  Partner with Us
                </a>
              </li>
              <li
                class="footer-link"
              >
                <a
                  href="/shipping"
                >
                  Delivery Info
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="footer-block"
        >
          <div
            class="block-title"
          >
            <h3
              class="text-uppercase"
            >
              Quick Links
            </h3>
          </div>
          <div
            class="block-content"
          >
            <ul>
              <li
                class="footer-link"
              >
                <a
                  href="/contact"
                >
                  Get in Touch
                </a>
              </li>
              <li
                class="footer-link"
              >
                <a
                  href="/sell"
                >
                  Partner with Us
                </a>
              </li>
              <li
                class="footer-link"
              >
                <a
                  href="/shipping"
                >
                  Delivery Info
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="footer-block"
        >
          <div
            class="block-title"
          >
            <h3
              class="text-uppercase"
            >
              Stay Updated
            </h3>
            <div
              class="newsletter-form"
            >
              <p>
                Subscribe to Our Newsletter
              </p>
              <form>
                <div
                  class="subscribe"
                >
                  <div
                    class="input-box inline-btn-box invalid"
                  >
                    <div
                      class="input-text-block"
                    >
                      <input
                        autocomplete="on"
                        class="input-text"
                        name="email"
                        placeholder="Enter Your Email Address"
                        type="text"
                        value="test@example.com"
                      />
                      <button
                        class="input-btn custom-btn-primary md text-only icon-left "
                        style="border-radius: 3px;"
                        type="submit"
                      >
                        <span
                          class="btn-text"
                        >
                          Join Now
                        </span>
                      </button>
                    </div>
                    <span
                      class="invalid-message"
                    >
                      I
                    </span>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer-copyright"
      >
        <span>
          Â© 2025 RMIT Store. All rights reserved.
        </span>
      </div>
      <ul
        class="footer-social-item"
      >
        <li>
          <a
            href="/#facebook"
            rel="noreferrer noopener"
            target="_blank"
          >
            <span
              class="facebook-icon"
            />
          </a>
        </li>
        <li>
          <a
            href="/#instagram"
            rel="noreferrer noopener"
            target="_blank"
          >
            <span
              class="instagram-icon"
            />
          </a>
        </li>
        <li>
          <a
            href="/#twitter"
            rel="noreferrer noopener"
            target="_blank"
          >
            <span
              class="twitter-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </footer>
</DocumentFragment>
`;
